test/e2e/alterBuiltManifest.ts

```diff
@@ -0,0 +1,92 @@
+/**
+ * Before this file was created, there was no way to pass runtime flags from an E2E test to the built extension.
+ * (Well okay, more precisely, you could change certain things through Fixtures, and you could change certain things
+ * through Ganache parameters, but both of these didn't load until late in the app load. Sentry is one of the first
+ * things to load, and it needed a way to get runtime flags very early in the app load.)
+ *
+ * The way this works is a bit of a hack, but it's the only solution we could come up with that would load early enough:
+ *
+ * 1) A global beforeEach hook in the E2E tests backs up the manifest.json file
+ * 2) The helpers.withFixtures() function calls setManifestFlags(), which reads in the manifest file and parses it
+ * 3) We alter the manifest with CircleCI environment variables and custom flags, then write it back to manifest.json
+ * 4) The test runs, and the built extension can call getManifestFlags() to get the custom flags
+ * 5) A global afterEach hook restores the backup copy of the manifest, so that the next test gets the normal manifest
+ */
+import fs from 'fs';
+import { ManifestFlags } from '../../app/scripts/lib/manifestFlags';
+
+const folder = `dist/${process.env.SELENIUM_BROWSER}`;
+
+// Global beforeEach hook to backup the manifest.json file
+if (typeof beforeEach === 'function' && process.env.SELENIUM_BROWSER) {
+  beforeEach(() => {
+    console.debug('alterBuiltManifest.ts -- beforeEach hook');
+
+    restoreBackupManifest();
+
+    backupManifest();
+  });
+}
+
+// Global afterEach hook to restore the backup manifest
+if (typeof afterEach === 'function' && process.env.SELENIUM_BROWSER) {
+  afterEach(() => {
+    console.debug('alterBuiltManifest.ts -- afterEach hook');
+
+    // create manifest.altered.json
+    backupManifest('altered');
+
+    restoreBackupManifest();
+  });
+}
+
+function backupManifest(newExtension = 'backup') {
+  fs.cpSync(
+    `${folder}/manifest.json`,
+    `${folder}/manifest.${newExtension}.json`,
+    {
+      preserveTimestamps: true,
+    },
+  );
+}
+
+function restoreBackupManifest() {
+  if (fs.existsSync(`${folder}/manifest.backup.json`)) {
+    fs.cpSync(`${folder}/manifest.backup.json`, `${folder}/manifest.json`, {
+      preserveTimestamps: true,
+    });
+  }
+}
+
+function parseIntOrUndefined(value: string | undefined): number | undefined {
+  return value ? parseInt(value, 10) : undefined;
+}
+
+/**
+ * Alter the manifest with CircleCI environment variables and custom flags
+ *
+ * @param flags - Custom flags to set
+ * @param flags.circleci - This will usually come in as undefined, and be set in this function
```

Comment:
If `flags.doNotForceSentryForThisTest` is set, and `process.env.CIRCLECI` is truthy, will `doNotForceSentryForThisTest` be set? I don't see how. Also, if `flags.circleci` is set when `process.env.CIRCLECI` is truthy flags.circleci will be altered.

Is this intentional behavior?

Reply:
Yes that's all intentional.

I'm testing it right now and when you call it with no argument `setManifestFlags()` it gets set to the default of `{}` and then if `process.env.CIRCLECI` then those variables get set.

If you call `setManifestFlags({doNotForceSentryForThisTest: true});` then it leaves that property alone and only alters the `circleci` property.  Any other property `foo` `bar` whatever will be left as is.

`flags.circleci` is not meant to be an input, thus it's overwritten.  Maybe there's a better way to structure and communicate that though?

Reply:
```
 is not meant to be an input, thus it's overwritten. Maybe there's a better way to structure and communicate that though?
```
 
Ah, I think you shouldn't define it in the input params then.